api/envoy/config/core/v3/substitution_format_string.proto

```diff
@@ -61,6 +64,12 @@ message SubstitutionFormatString {
     //  }
     //
     google.protobuf.Struct json_format = 2 [(validate.rules).message = {required: true}];
+
+    // Specify a format with command operators, similar to text_format, but from a DataSource.
+    DataSource text_format_source = 5;
+
+    // Specify fixed text with no command operators.
+    string fixed_text = 6;
```

Comment:
what is the purpose of this format?  Especially how it is used in file_access_log

Reply:
That's a good question and I realize now that I forgot to call out the original motivation for this PR. I linked https://github.com/envoyproxy/envoy/pull/14221 as context.

The idea here is that we want a generic way to define HTTP response body rewrites. One interesting use case is to replace a non-empty HTTP body with an empty one, or one with a fixed string, like "denied".

One thing your comment exposes is that we're overloading the purpose of this object for generic rewrites and file_access_log formatting. There's probably an opportunity here to create some shared objects that don't cross concerns as much, making this less confusing.

Reply:
Yeah as I said in #14221, I prefer just relax the text format validation. Will we see performance improvement of this formatter over the normal text_format?

Reply:
Great. I relax the format validation and remove fixed_text

Reply:
This is polluting SubstitutionFormatString class.  I see 1) `DataSource` of `text_format` and 2) `fixed_text` should not be part of this SubstitutionFormatString class.  They should be in the ResponseMapper message in https://github.com/envoyproxy/envoy/pull/14221

Reply:
While I think it could be correct to specify a `DataSource` field in the `ResponseMapper` message in #14221, I also think it's reasonable and correct to allow for the `text_format` to come from a `DataSource` on `SubstitutionFormatString`, since a `DataSource` supports basic strings (via `inline_string`). The semantics of `SubstitutionFormatString` remain the same, we just have a behavioral extension that allows it to come from a source, which is nice. What do you think about the tradeoff of having the additional behavior here versus having users of this class define their own DataSource field to replace it?

Reply:
I think there's also a reasonable argument to be made that the implementation of a `SubstitutionFormatString` using a `DataSouce` field was really straightforward. To move the source up a layer of abstraction, we'd need the option of creating a format string by telling it where it is sourced from. This conflicts with the existing fields in the format string that tell it where it is sourced from (i.e. text_format and json_format).

Reply:
It is an interesting idea of replacing `json_format` with `text_format`.

Reply:
Now that I think about it, though, `json_format` strings are escaped for json while `text_format` strings are not. That's fairly important and prevents us from totally getting rid of the `json_format` field in favor of `text_format` until we have a way to guarantee that json-as-text can be escaped nicely. Maybe we could add/extend the command operator(s) to support escaping?